player/weapons/revolver/revolver.gd

```diff
@@ -0,0 +1,75 @@
+class_name Revolver
+extends Weapon
+
+const Scene = preload("res://player/weapons/revolver/revolver.tscn")
+
+var damage: float = 1.0
+var base_fire_rate: float = 0.75
+var reload_time: float = 0.35
+
+var _magazine: PlayerMagazine
+var _is_reloading: bool = false
+var _is_shooting: bool = false
+
```

Comment:
i'm thinking we could have a `_mode` or something, to indicate if it's semi-auto or auto, then for semi-auto weapons we'd need to make sure the "trigger" was released before allowing to fire again. then we wouldn't have to worry about it in the derived classes.

so, for semi-auto, when we "start_shooting", we flag an "is_reset" (just an example name) bool to false, and it only gets set to true when "stop_shooting" is called (when the trigger is released)

```gdscript
func _can_fire() -> bool:
  # other early out checks on magazine shots left etc

  if _mode == 'semiauto':
    return _is_trigger_pulled and _is_reset

  if _mode == 'auto':
    return _is_trigger_pulled
```